### PR TITLE
Add page about gradle setup

### PR DESCRIPTION
**Pull request by writtmeyer**
This new page is part of the basics section of the site. It partially addresses issue #69.

The mentioning of the current stable and beta versions in the sidebar is not part of this PR and will be done in a separate PR to keep the PR discussion focused.

### CODE REVIEW

File: basics/gradle.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+layout: content
+title: Gradle Integration
+author: wolfram
+as_version: 3.0
+cl_version: 1.0.2 or 1.1.0-beta5
+order: 6
+---
+
+### Top level build configuration
+You can add ConstraintLayout as you do with other dependencies: By adding it to your gradle `build.xml` file.
```
Comment (rock3r) [inline basics/gradle.md]: `By` shouldn't have a capital `B`, it doesn't follow a full stop.

File: basics/gradle.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+}
+
+allprojects {
+  repositories {
+    google()
+    jcenter()
+  }
+}
+```
+
```
Comment (rock3r) [inline basics/gradle.md]: Nit: extra whiteline

File: basics/gradle.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+With this configuration in place, you simply add the version of ConstraintLayout you want to the module specific gradle file (usually within the `app` module):
+
+```gradle
+dependencies {
+  // other libs...
+  implementation 'com.android.support.constraint:constraint-layout:1.1.0-beta5'
+  // other libs...
+}
+```
+
```
Comment (rock3r) [inline basics/gradle.md]: Nit: extra whiteline

File: basics/gradle.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+All versions are listed on Google's [maven repository](https://dl.google.com/dl/android/maven2/com/android/support/constraint/group-index.xml).
+
+Features of the 1.1 branch that are missing in the 1.0 release:
+* [Barriers](barriers.html)
+* [Placeholders](https://developer.android.com/reference/android/support/constraint/Placeholder.html)
+* [Groups](https://developer.android.com/reference/android/support/constraint/Group.html)
+* [Circular positioning](https://developer.android.com/reference/android/support/constraint/ConstraintLayout.html#CircularPositioning)
+* Percent dimensions
+
```
Comment (rock3r) [inline basics/gradle.md]: Nit: extra whiteline

File: basics/gradle.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+### Android Studio's Layout Editor
+Newer features of the ConstraintLayout are best supported in newer versions of Android Studio. While the 3.0.x versions are stable, they might not support all features of the latest betas. On the other hand the 3.1 and 3.2 versions are in active development. They are bound to contain bugs and might not properly support all of the older features.
+
+If you use XML anyway, the specific version of Android Studio is of less importance.
+
+
```
Comment (rock3r) [inline basics/gradle.md]: Nit: extra whitelines

File: basics/gradle.md (hunk 1 above)
Comment (writtmeyer) [reply basics/gradle.md]: Didn't know that. In Germany it's the other way round: Always use capital letters after colons. Will change that.

**Comment by writtmeyer** [comment]
I agree, That was my first thought when seeing the issue. We could of course do some JavaScript parsing of the maven repo's xml - but I'd prefer not to do that :-)

File: basics/gradle.md (hunk 2 above)
Comment (writtmeyer) [reply basics/gradle.md]: I did use all those extra line breaks on purpose (apart from those at the end of the file). I consider the document to be easier readable for the editor with two line breaks where a new section starts. I always do that - unless it causes additional <p> or <br> tags to appear in the final document. But, if we use a different style here on constraintlayout.github.io, I'm fine with that and will remove them.

File: basics/gradle.md (hunk 1 above)
Comment (rock3r) [reply basics/gradle.md]: Yeah some languages have "non-standard" rules when it comes to typography. I knew for example French puts a space before some punctuation (e.g., question marks and exclamation marks) but in English you generally don't do that

**Pull request merged**